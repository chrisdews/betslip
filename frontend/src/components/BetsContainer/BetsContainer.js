import React from "react";
import PropTypes from "prop-types";
import BetInfo from "../BetInfo";

function BetsContainer({ bets, betBuilder }) {
  return bets ? (
    <div>
      {bets.map((bet) => (
        <BetInfo
          key={bet.betId}
          betName={bet.name}
          bestOdds={bet.bestOdds}
          betBuilder={betBuilder}
        />
      ))}
    </div>
  ) : (
    "loading"
  );
}

BetsContainer.propTypes = {
  bets: PropTypes.array,
  betBuilder: PropTypes.func,
};

export default BetsContainer;
