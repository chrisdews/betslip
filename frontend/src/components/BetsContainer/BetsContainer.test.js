import React from "react";
import BetsContainer from "./BetsContainer";
import { render } from "@testing-library/react";

describe("BetInfo", () => {
  it("renders correctly - snapshot", () => {
    const { container } = render(<BetsContainer />);
    expect(container).toMatchSnapshot();
  });
});