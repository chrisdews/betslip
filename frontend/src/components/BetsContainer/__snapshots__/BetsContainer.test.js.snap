// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BetInfo renders correctly - snapshot 1`] = `
<div>
  <div
    class="sc-eCApnc fupMKc"
  >
    <h3>
      loading...
    </h3>
    <img
      class="sc-hKFxyN eaodwR"
      src="oc-logo.png"
    />
  </div>
</div>
`;
